src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains the ticking function that is responsible for invoking block acquisition functionality and executing it in periodic fashion.
+
+
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  ) where
+
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Block (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.Text.Internal
+    ( showText )
+import Formatting
+    ( sformat, (%) )
+
+import qualified Formatting as F
+
+
+blockSyncerBracket
+    :: IO [Block]
+    -- ^ a way to get a new block
+    -> (Block -> IO ())
+    -- ^ action taken on a new block
+    -> Int
+    -- ^ tick time in seconds
+    -> IO ()
+blockSyncerBracket getNextBlocks action tickTime =
+    bracket
+      (return ())
+      (\_->return ())
+      (tickingFunction getNextBlocks action tickTime)
+
+tickingFunction
+    :: IO [Block]
+    -> (Block -> IO ())
+    -> Int
+    -> ()
+    -> IO ()
+tickingFunction getNextBlocks action tickTime _ =
+    go `catch`
+    (\(e :: SomeException) ->
+            let msg = "Terminating tickingFunction of BlockSyncer due to " % F.shown
+            in (putStrLn . showText) $ sformat msg e
+    )
+    where
+      go :: IO ()
+      go = do
+          blocksToProcess <- getNextBlocks
+          mapM_ action blocksToProcess
```

Comment:
We miss a crucial part here which is: controlling that those blocks are new blocks. So, maybe the requirements weren't really clear but, here's the context:

- We have a networking layer that allows us to fetch the latest block from the network. It doesn't do much more than, fetching the tip; there's no particular logic involved apart from implementing the right communication protocol (e.g. an http request to the bridge). 

- Then, the ticking function contains an extra level of logic where, we want to know if we need to process the current block. Maybe we already have, and we do not want to process it. So, the action should only be called with yet unprocessed blocks.

Reply:
done, now tickingFunction is responsible not to take action on the block it previously consumed. I added the comment also specifying known limitation (lack of handling block gaps)